File: app/src/main/java/com/live/timertest/MyTimer.java
package com.live.timertest;

interface onListener {
    public void onComplete();
}
public class MyTimer {
    MyTimer(String name, boolean isDaemon)
    {

    }

    public void schedule(long delay, onListener listener )
    {
        try {
            Thread.sleep(delay);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        if( listener != null )
            listener.onComplete();
    }

}

File: app/src/main/java/com/live/timertest/MyTimer.kt
package com.live.timertest

interface onListener {
    public fun onComplete();
}

open class MyTimer(name: String, boolean: Boolean) {
    fun schedule(delay: Long, listener: onListener )
    {
        try {
            Thread.sleep(delay);
        } catch (e:InterruptedException) {
            e.printStackTrace();
        }

        if( listener != null )
            listener.onComplete();
    }
}